src/resources/Catalogs/CatalogInterfaces.ts

```diff
@@ -3,10 +3,37 @@ export interface CatalogsListOptions {
     pageSize?: number;
 }
 
+export interface CatalogFieldsMapping {
+    [name: string]: string;
+}
+
+export interface CreateCatalogConfigurationModel {
+    id: string;
+    name: string;
+    product: Omit<CreateProductHierarchyModel, 'fields'>;
+    variant?: Omit<CreateVariantHierarchyModel, 'fields'>;
+    availability?: Omit<CreateAvailabilityHierarchyModel, 'fields'>;
+    fieldsMappings: CatalogFieldsMapping;
+}
+
+export interface CatalogConfigurationModel {
+    id: string;
+    name: string;
+    product: HierarchyWithFields<ProductHierarchyModel>;
+    variant?: HierarchyWithFields<VariantHierarchyModel>;
+    availability?: HierarchyWithFields<AvailabilityHierarchyModel>;
+    fieldsMappings: CatalogFieldsMapping;
+}
+
 export interface BaseCatalogModel {
     id: string;
     name: string;
     description?: string;
+    sourceId?: string;
```

Comment:
`sourceId` ~is~ will not be optional, do we want to make it manual here to avoid breaking later?

Reply:
If we put it mandatory here, every object that needs `BaseCatalogModel` *needs* to define it, which is a breaking change :(